tests/django_pg_migration_tools/test_operations.py

```diff
@@ -260,3 +260,170 @@ def test_when_not_allowed_to_migrate(self):
+        with connection.cursor() as cursor:
+            cursor.execute(
+                _CHECK_INDEX_EXISTS_QUERY,
+                {"table_name": "example_app_charmodel", "index_name": "char_field_idx"},
+            )
+            assert cursor.fetchone() is None
+
+        # Prove that the lock_timeout has been set back to the default (1s)
+        with connection.cursor() as cursor:
+            cursor.execute(
+                operations.SaferRemoveIndexConcurrently.SHOW_LOCK_TIMEOUT_QUERY
+            )
+            assert cursor.fetchone()[0] == "1s"
+
+        # Assert on the sequence of expected SQL queries:
+        assert queries[0]["sql"] == "SHOW lock_timeout;"
+        assert queries[1]["sql"] == "SET lock_timeout = 0;"
+        assert queries[2]["sql"] == 'DROP INDEX CONCURRENTLY IF EXISTS "char_field_idx"'
+        assert queries[3]["sql"] == "SET lock_timeout = '1s';"
+
+        # Reverse the migration to re-create the index and verify that the
+        # lock_timeout queries are correct.
+        with connection.schema_editor(atomic=False, collect_sql=False) as editor:
+            with utils.CaptureQueriesContext(connection) as reverse_queries:
+                operation.database_backwards(
+                    self.app_label, editor, new_state, project_state
+                )
+
+        assert reverse_queries[0]["sql"] == "SHOW lock_timeout;"
+        assert reverse_queries[1]["sql"] == "SET lock_timeout = 0;"
+        assert reverse_queries[2]["sql"] == dedent("""
+            SELECT relname
+            FROM pg_class, pg_index
+            WHERE (
+                pg_index.indisvalid = false
+                AND pg_index.indexrelid = pg_class.oid
+                AND relname = 'char_field_idx'
+            );
+            """)
+        assert (
+            reverse_queries[3]["sql"]
+            == 'CREATE INDEX CONCURRENTLY IF NOT EXISTS "char_field_idx" ON "example_app_charmodel" ("char_field")'
+        )
+        assert reverse_queries[4]["sql"] == "SET lock_timeout = '1s';"
+
+    # Disable the overall test transaction because a concurrent index cannot
+    # be triggered/tested inside of a transaction.
+    @pytest.mark.django_db(transaction=True)
+    @override_settings(DATABASE_ROUTERS=[NeverAllow()])
+    def test_when_not_allowed_to_migrate(self):
+        # Prove that the index exists before running the removal operation.
+        with connection.cursor() as cursor:
+            cursor.execute(
+                _CHECK_INDEX_EXISTS_QUERY,
+                {"table_name": "example_app_charmodel", "index_name": "char_field_idx"},
+            )
+            assert cursor.fetchone()
+
+        project_state = ProjectState()
+        project_state.add_model(ModelState.from_model(CharModel))
+        new_state = project_state.clone()
+
+        operation = operations.SaferRemoveIndexConcurrently(
+            "charmodel",
+            "char_field_idx",
+        )
+        # Proceed to try and add the index:
```

Comment:
copy/paste: "remove" instead of "add"

Reply:
Thx: 54d6cf9d4a37adffb3b0a4b70b7b301236f5cf93